be/src/vec/olap/vertical_block_reader.cpp

```diff
@@ -198,13 +198,23 @@ void VerticalBlockReader::_init_agg_state(const ReaderParams& read_params) {
         });
         _agg_places.push_back(place);
 
-        // calculate `has_string` tag.
-        _stored_has_string_tag[idx] =
+        // calculate `_has_variable_length_tag` tag. like string, array, map
+        _stored_has_variable_length_tag[idx] =
                 _stored_data_columns[idx]->is_column_string() ||
                 (_stored_data_columns[idx]->is_nullable() &&
                  reinterpret_cast<ColumnNullable*>(_stored_data_columns[idx].get())
                          ->get_nested_column_ptr()
-                         ->is_column_string());
+                         ->is_column_string()) ||
```

Comment:
we should add a function is_variable_length() to IColumn and return true in ColumnString/Array/Map and any other variable length column.

Reply:
ColumnNullable call nested.is_variable_length()